chall02/dapalmer.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/python
+
+import sys
+
+def main():
+    if len(sys.argv) == 2:
+        morseCode = {
+            'A': '.-',
+            'B': '-...',
+            'C': '-.-.',
+            'D': '-..',
+            'E': '.',
+            'F': '..-.',   
+            'G': '--.',
+            'H': '....',
+            'I': '..',   
+            'J': '.---',
+            'K': '-.-',
+            'L': '.-..',   
+            'M': '--',
+            'N': '-.',
+            'O': '---',   
+            'P': '.--.',
+            'Q': '--.-',
+            'R': '.-.',
+            'S': '...',
+            'T': '-',
+            'U': '..-',
+            'V': '...-',
+            'W': '.--',
+            'X': '-..-',
+            'Y': '-.--',
+            'Z': '--..',
+            ' ': ' '       
+        }
+        origString = sys.argv[1].upper()
+        morseString = ''
+
+        for char in origString:
+            if char >= 'A' and char <= 'Z' or char == ' ':
+                morseString += morseCode[char]
+        print morseString
```

Comment:
Your code will crash if there are un expected characters in the string.

Also you should use comprehension to make your code more pythonic and clearer.

```suggestion
        if not origString.replace(" ", "").isalpha():
            print('usage:', sys.argv[0], '<a-zA-Z string>')
        print("".join(morseCode[c] for c in origString))
```